### PR TITLE
clarify -1 for adapter_response

### LINKED ISSUE #3145: Clarify -1 value for adapter_response
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

[Per slack thread](https://dbt-labs.slack.com/archives/C017GDLAF7D/p1680682398935539), users have reached out confused about what the `-1` means when they see `'rows_affected': {'_message': 'CREATE VIEW', 'code': 'CREATE VIEW', 'rows_affected': -1}}` in their run results.json artifact. 

`-1` is the standard value for `rowcount` in cases where your query's rowcount can't be determined: https://peps.python.org/pep-0249/#rowcount

Exactly as Benoit said, creating a view is just saving a query for the future — it doesn't actually create/alter/return any rows until you actually select from it. 

### What part(s) of the page would you like to see updated?

any page where adapter_response is discussed (which could deserve its own page)

- https://docs.getdbt.com/reference/artifacts/run-results-json#top-level-keys
- https://docs.getdbt.com/reference/dbt-classes#result-objects

### Additional information

hey @dbeatty10 - if you have any bandwidth at all in the next few weeks, can i create a pr and as for your technical advice so i can explain this correctly and also make sure I'm adding content in the correct places?

### PR DESCRIPTION
**Pull request by mirnawong1**
adding more info on what `-1` means when they see it in their run results.json artifact. 

More info in this [slack thread](https://dbt-labs.slack.com/archives/C017GDLAF7D/p1680682398935539)

Resolves #3145

### CODE REVIEW

**Comment by mirnawong1** [comment]
hi @dbeatty10 ! you may remember i asked if you could kindly review a pr related to `-1` when users see this in their run results. I've created the pr and could you please review it when you have a moment for technical accuracy? I've added the content using a partial in a few places.  

deploy previews are as follows:

[run results artifacts page](https://deploy-preview-3762--docs-getdbt-com.netlify.app/reference/artifacts/run-results-json#top-level-keys)
[sources artifacts page](https://deploy-preview-3762--docs-getdbt-com.netlify.app/reference/artifacts/sources-json#top-level-keys)
[dbt Classes page](https://deploy-preview-3762--docs-getdbt-com.netlify.app/reference/dbt-classes#result-objects)

File: website/snippets/_run-result.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- `adapter_response`: Dictionary of metadata returned from the database, which varies by adapter. For example, success `code`, number of `rows_affected`, total `bytes_processed`, and so on. Not populated by tests, as of v0.19.0; we plan to fix in a future release ([dbt#2580](https://github.com/dbt-labs/dbt-core/issues/2580)).
```
Comment (dbeatty10) [inline website/snippets/_run-result.md]: Can we drop the explicit call-out related to v0.19.0 and replace it with a general statement instead? We closed ([dbt#2580](https://github.com/dbt-labs/dbt-core/issues/2580)) as stale, and I don't think we implemented it.

```suggestion
- `adapter_response`: Dictionary of metadata returned from the database, which varies by adapter. For example, success `code`, number of `rows_affected`, total `bytes_processed`, and so on. Not applicable for [tests](/docs/build/tests).
```

File: website/snippets/_run-result.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+    * `rows_affected` returns the number of rows modified by the last statement executed. In cases where the query's row count can't be determined or isn't applicable (such as when creating a <Term id="view">view</Term>), a [standard value](https://peps.python.org/pep-0249/#rowcount) of `-1` is returned for `rowcount`.
```
Comment (dbeatty10) [inline website/snippets/_run-result.md:2]: 🤩

**Review by dbeatty10 — APPROVED** [review]
This looks great @mirnawong1.

Left one optional suggestion.

File: website/snippets/_run-result.md (hunk 1 above)
Comment (mirnawong1) [reply website/snippets/_run-result.md]: omg yes thank you @dbeatty10 ! sugg committed woohoo!

**Pull request merged**